File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) 
    {
        int minLength = Integer.MAX_VALUE;
        int i = 0, sum = 0;
        for(int j = 0; j < nums.length; j++)
        {
            sum += nums[j];
            while(sum >= target)
            {
                minLength = Math.min(j - i +1, minLength);
                sum -= nums[i];
                i++;
            }
        }
        return minLength == Integer.MAX_VALUE ? 0 : minLength;
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& a) 
    {
        int minLength=1e9;
        int i=0,sum=0;
        for(int j=0;j<a.size();j++)
        {
            sum+=a[j];

            while(sum>=target)
            {
                minLength=min(minLength,j-i+1);
                sum-=a[i];
                i++;
            }
        }
        return (minLength==1e9)?0:minLength;
    }
};